subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java

```diff
@@ -272,6 +274,13 @@ private <T> Provider<T> getToolchainTool(Project project, BiFunction<JavaToolcha
         return toolMapper.apply(service, extension.getToolchain());
     }
 
+    private Provider<JavaLauncher> getJavaLauncher(Project project) {
+        final JavaPluginExtension extension = extensionOf(project, JavaPluginExtension.class);
+        final JavaToolchainService service = extensionOf(project, JavaToolchainService.class);
+        return service.launcherFor(extension.getToolchain())
+            .map(it -> ((JavaToolchain) it.getMetadata()).isFallbackToolchain() ? null : it);
```

Comment:
Same as for GroovyCompile: I suppose we need to follow up on this, right?

Reply:
Yep, that's right